armi/reactor/blueprints/assemblyBlueprint.py

```diff
@@ -174,15 +263,16 @@ def _createBlock(self, cs, blueprint, bDesign, axialIndex):
 
     def _checkParamConsistency(self, a):
         """Check that the number of block params specified is equal to the number of blocks specified."""
-        materialInputs = self.materialModifications or {}
+        materialInputs = self._compileAssemblyMaterialInput()
```

Comment:
It's surprising that a checkParamConsistency call would do data compilation step. I suggest reverting this line and calling `self._compileAssemblyMaterialInput()` at the higher level, right before `_checkParamConsistency` is called.

Reply:
I got rid of the `_compileAssemblyMaterialInput` method and changed this part of the code to just pull directly from `self.materialModifications` as needed.